packages/image_picker/ios/Classes/ImagePickerPlugin.m

```diff
@@ -99,11 +99,60 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
                                    details:nil]);
         break;
     }
+  } else if ([@"generateImageThumbnail" isEqualToString:call.method]) {
+    NSString *imagePath = [call.arguments objectForKey:@"originalImagePath"];
+    NSNumber *width = [call.arguments objectForKey:@"width"];
+    NSNumber *height = [call.arguments objectForKey:@"height"];
+
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+      UIImage *image = [UIImage imageWithContentsOfFile:imagePath];
+
+      [self generateThumbnail:image result:result width:width height:height];
+    });
+  } else if ([@"generateVideoThumbnail" isEqualToString:call.method]) {
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+      NSString *imagePath = [call.arguments objectForKey:@"originalVideoPath"];
+      NSNumber *width = [call.arguments objectForKey:@"width"];
+      NSNumber *height = [call.arguments objectForKey:@"height"];
+
+      NSURL *videoURL = [NSURL fileURLWithPath:imagePath];
+      MPMoviePlayerController *player =
+          [[MPMoviePlayerController alloc] initWithContentURL:videoURL];
+      UIImage *image = [player thumbnailImageAtTime:1.0
+                                         timeOption:MPMovieTimeOptionNearestKeyFrame];
+
+      [self generateThumbnail:image result:result width:width height:height];
+    });
   } else {
     result(FlutterMethodNotImplemented);
   }
 }
 
+- (void)generateThumbnail:(UIImage *)image
```

Comment:
What is the difference of this and the regular pickImage with maxWidth and maxHeight?